### ISSUE TITLE
Option to attempt repair on corrupted files

### ISSUE THREAD

**Issue by K2M74** [Yutaka-Sawada/MultiPar#74]
When restoring, the TMP file will be generated first and then repaired. The following problems exist in the processing of large files:

For SSD hard drives, additional writes are added.
For HDD drives, simultaneous reading and writing can greatly extend the repair time.
For all hard drives, the same additional space is required for recovery.

Imagine recovering a single large file of 100+GB.....

It may be possible to modify the recovery directly on the original file, for example Bittorrent software checks and downloads the missing parts of the file, which can be an option.

I don't speak English, and the translation software I use may make mistakes.

**Comment by Yutaka-Sawada** [comment]
> It may be possible to modify the recovery directly on the original file

No, it won't for a file recovery sometimes. In the referred Bittorrent system, file data is splitted into many pieces. Then, it can collect good pieces to construct the original file. It's same as joining a new file from many small files.

1) There are many good and bad pieces over network.
2) Bittorrent picks good pieces and joins them to make a good file.

For example, there are 4 pirces.
Bittorrent: [A],[B],[C],[D] -> join -> [ABCD]

Now, imagine recovering a single large file of 100+GB.....

1) There is a damaged file of 100 GB.
2) For example, 99 GB of data is complete.
3) It creates a temporary file of 100 GB newly, and copy complete data.
4) Parchive recovers 1 GB lost data, and put them on the temporary file.

For example, there are 3 complete blocks and 1 parity block.
Parchive : [ABD] + [P] -> copy -> [AB_D] -> restore -> [ABCD]

The difference is location of original data. Bittorrent gets data from other PCs over network. Parchive reads data on the same PC. It's difficult to restore a damaged file directly by over-writing corrected data on damaged data. This is because, most file system doesn't support inserting data in the middle of a file.

In the above sample, there is a damaged file [ABD]. It cannot insert the recovered block [C] between [AB] and [D]. To re-construct the original file, it needs to make a temporary file [AB_D] at first. Then, it can put block [C] on the spaced position of [AB_D].

Only when damage didn't change position of complete blocks, direct recovery will be possible. But, there is a risk of failed recovery. Over-writing data would erase original data. If you understand and accept the risk, I may try to implement the recovering method. It will work on the limited case.

**Comment by K2M74** [comment]
@Yutaka-Sawada 
I understand that files of different sizes, the original file modification will indeed have problems, this is also my suggestion to provide optional options instead of default.
There should be no problem in restoring files of the same size. For example: ABCD is damaged into ABXD. At this time, the data size of ABXD = ABCD, and you only need to replace X
The example of Bittorrent is the case. The file is stored locally: ABXD. After Bittorrent check, it is found that X is wrong.
Bittorrent is downloaded to the correct C by other peers on the network and overwrites X. At this time, the hard disk is not written on a large scale.
Through the verification check, the software can determine the damaged block, and only need to modify this part of the data. If the file system does not support it, it can be forced to generate a TMP file for processing. At least NTFS is supported.

The [Data degradation](https://en.wikipedia.org/wiki/Data_degradation) of the hard disk is very common, it does not change the file size and data location, just a few bits are wrong.

Of course, I don't know how to write code, and I don't understand the difficulty, so try your best.

**Comment by Yutaka-Sawada** [comment]
I made a sample of your requested feature. It won't make temporary files at recovery. Be careful the pros and cons. I wrote details on ReadMe.txt in the sample package.

I put the sample (multipar_sample_2022-10-08.zip) in "MultiPar_sample" folder on [OneDrive](https://1drv.ms/u/s!AtGhNMUyvbWOg0cF2UHcs709Icv4?e=UUj7bE). If there is no serious problem, I will release the feature in next version. Then, more users may test it on their usage.

By the way, how is the naming ? I set the name "Aligned verification" now, because alignment of blocks is important. Is it natural words in English ? Other possible words may be "Fast verification" or "Orderly verification". I don't know nuance of English words.
